src/cxx_support_test_cpp.h

```diff
@@ -0,0 +1,55 @@
+// Copyright 2022 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+// Build in test mode only to test cxx integration.
+
+#include <cstdint>
+#include <memory>
+
+#pragma once
+
+const uint8_t UNINITIALIZED = 0;
+const uint8_t INITIALIZED = 1;
+const uint8_t METHOD_CALLED = 2;
+
+class Foo {
+public:
+    Foo() {
+        data[0] = INITIALIZED;
+    }
+    uint8_t get_status() const {
+        return data[0];
+    }
+    void modify() {
+        data[0] = METHOD_CALLED;
+    }
+private:
+    uint32_t data[4]; // exactly match layout declared in Rust.
+};
+
+inline Foo* create_uninitialized_foo() {
+    std::allocator<Foo> alloc;
+    Foo* data = alloc.allocate(1);
+    (reinterpret_cast<uint32_t*>(data))[0] = UNINITIALIZED;
```

Comment:
This is a strict aliasing violation. Instead, consider:

```cpp
std::memcpy(data, &kUninit, 1);
```

Reply:
Done